articles/embedded.md

```diff
@@ -0,0 +1,309 @@
+
+*Meta-Note*: Please only add a short description here, linking to 
+more detailed pages if necessary.
+
+### A-RTOS: *RTOS Proof-of-Concept* 
+
+#### Questions: *Q-RTOS*
+
+Provide a proof-of-concept RTOS. In the OFERA project, we chose 
+[NuttX](http://nuttx.org/), because it is largely POSIX compatible 
+and thus eases porting. There are also experiments based on 
+[RIOT](https://www.riot-os.org/) (cf. 
+[github.com/astralien3000/riot-ros2](https://github.com/astralien3000/riot-ros2))
+and FreeRTOS (cf. 
+[github.com/ros2/ros2_embedded_freertos](https://github.com/ros2/ros2_embedded_freertos)) |
+
+### A-BUILD-META: *Meta-Build*
+
+#### Questions: Q-BUILD 
+
+Explores a meta-build approach to transform ROS 2 CMakeLists.txt to 
+RTOS-specific build instructions. |
+
+###  A-BUILD-NUTTX: *NuttX-specific build*
+
+#### Questions: Q-BUILD 
+
+OFERA has integrated (parts of) micro-ROS directly as an app and 
+as a library in the NuttX build.
+
+### A-BUILD-ARDUINO: *Arduino Build*
+
+### Questions: Q-BUILD  
+
+ROBOTIS has explored building all the libraries using the Arduino 
+IDE, cf. [github.com/ROBOTIS-GIT/OpenCR](https://github.com/ROBOTIS-GIT/OpenCR/tree/feature-ros2-micrortps). 
+This required some manual changes and thus does not scale, but 
+can get you off the ground.
+
+###  A-NUTTX-LIBCXX: *C++11/higher support for NuttX*
+
+#### Questions: Q-LANG  
+
+Build libxx from the LLVM project on NuttX, as a pre-requisite to building rclcpp.
+
+### A-PERF-RCLCPP-RESOURCE: *Determine resource use of rclcpp*
+
+#### Questions: Q-PERF 
+
+There are doubts whether rclcpp can really fit tiny MCUs, but we'll only 
+know once we tried and measured it.
+
+### A-DDS-XRCE: *Use of DDS-XRCE standard protocol*
+
+#### Questions: Q-COMM  
+
+This OMG standard defines a protocol for use by microcontrollers to 
+publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA 
+and ROBOTIS have demonstrated that it is a suitable protocol to seamlessly
+communicate with microcontrollers. See [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/)
+
+### A-Micro-XRCE-DDS: *Micro-XRCE Middleware usage*
+
+#### Questions: Q-COMM  
+
+OFERA, Robotis and others have integrated Micro XRCE-DDS middleware as 
+part of their solutions. This middleware provides an implementation of 
+DDS-XRCE standard. Integrations have been done on top of different RTOSs, 
+NuttX and FreeRTOS and using different underlying transports. 
+
+See [Micro XRCE-DDS](https://github.com/eProsima/Micro-XRCE-DDS) |
+
+## Prior and on-going works
+
+### ROS2-based approaches
+*   [**EU project OFERA**](http://ofera.eu/): The EU project OFERA 
+(Open Framework for Embedded Robot Applications) aims at a ROS 2-compatible 
+stack for MCUs in the range of STM32F4 or STM32L1, i.e. with possibly less 
+than 100kB RAM. The project partners currently investigate of using the 
+ROS 2 rmw, rcl and rclcpp layers as-is on the Micro XRCE-DDS implementation 
```

Comment:
Strange that the project documents and this document don't mention this investigation up front and instead start from the assumption of needing to create a separate, "compatible" implementation of the core ROS2 libraries. I'd like to see the outcome of this work and in particular numbers showing that it's just not possible to get rmw and rcl down to the tiny scale embedded size before I agree that *the* approach for ROS2 for these devices should be a separate implementation of the core libraries.